test/utils/Bytes.test.js

```diff
@@ -111,6 +111,174 @@ describe('Bytes', function () {
         });
       }
     });
+
+    describe('splice(bytes, bytes)', function () {
+      const replacement = ethers.toUtf8Bytes('REPLACEMENT');
+
+      it('replace from start', async function () {
+        const buffer = new Uint8Array(lorem);
+        const expected = new Uint8Array(buffer.length);
+        expected.set(replacement.slice(0, Math.min(replacement.length, buffer.length)), 0);
+        expected.set(
+          buffer.slice(Math.min(replacement.length, buffer.length)),
+          Math.min(replacement.length, buffer.length),
+        );
+
+        await expect(this.mock.$splice(lorem, ethers.Typed.bytes(replacement))).to.eventually.equal(
+          ethers.hexlify(expected),
+        );
+      });
+
+      it('replacement longer than buffer', async function () {
+        const longReplacement = ethers.toUtf8Bytes('THIS IS A VERY LONG REPLACEMENT THAT EXCEEDS THE BUFFER SIZE');
```

Comment:
Btw:
Length of `longReplacement` and `buffer`.
`{ long: 60, buffer: 123 }`

Reply:
Nice catch!

```suggestion
        const longReplacement = ethers.toUtf8Bytes('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi placerat elit non felis scelerisque faucibus. Donec eget blandit.');
```